### PR TITLE
fix(components): [el-message] & [el-notification] added missing props

### PR DESCRIPTION
**Pull request by codespikey**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by codespikex** [comment]
This PR fixes the issue for #4012 where the appendTo is missing in component props.

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -10338,7 +10338,7 @@ packages:
       stylus:
         optional: true
     dependencies:
-      esbuild: 0.13.12
+      esbuild: 0.13.13
```
Comment (sxzz) [inline pnpm-lock.yaml:10341]: please remove Irrelevant changes

File: packages/components/message/src/message.ts [hunk 2]
```diff
@@ -54,6 +54,10 @@ export const messageProps = buildProps({
     type: Number,
     default: 0,
   },
+  appendTo: {
+    type: definePropType<string | HTMLElement>([String, HTMLElement]),
```
Comment (sxzz) [inline packages/components/message/src/message.ts:58]:
```suggestion
    type: definePropType<string | HTMLElement>([String, Object]),
```

**Comment by sxzz** [comment]
Please add mocks for unit test, since there's seems no `HTMLElement` in `jest-dom`.
Also, you should refactor to latest dev branch.

**Comment by codespikex** [comment]
The ssr test is failing due to jest's node environment. I was able to mock the HTMLElement with this code
```javascript
if (global && !('HTMLElement' in global)) {
    Object.defineProperty(global, 'HTMLElement', {
      value: class HTMLElement {},
    })
  }
```
but this mock only works if used inside the 'packages/element-plus/\_\_tests\_\_/ssr.spec.ts' file.

**Comment by msidolphin** [comment]
appendTo is not a props, this is provided for API use

**Comment by codespikex** [comment]
Actually i can just remove it from props before passing the props to the component. 🤔

**Review by msidolphin — APPROVED** [review]

**Pull request closed without merging**